src/onnx/parse_matmul.cpp

```diff
@@ -91,51 +185,21 @@ struct parse_matmul : op_parser<parse_matmul>
             auto s1_lens        = a1->get_shape().lens();
             instruction_ref ba0 = a0;
             instruction_ref ba1 = a1;
+
             // try broadcasting if dimensions other than last two do not match
-            if(not std::equal(
-                   s0_lens.rbegin() + 2, s0_lens.rend(), s1_lens.rbegin() + 2, s1_lens.rend()))
+            if(is_quant_dot)
             {
-                auto l0_it = s0_lens.begin() + s0_lens.size() - 2;
-                std::vector<std::size_t> l0_broadcasted_lens(s0_lens.begin(), l0_it);
-                auto l1_it = s1_lens.begin() + s1_lens.size() - 2;
-                std::vector<std::size_t> l1_broadcasted_lens(s1_lens.begin(), l1_it);
-                auto output_lens =
-                    compute_broadcasted_lens(l0_broadcasted_lens, l1_broadcasted_lens);
-                l0_broadcasted_lens = output_lens;
-                l0_broadcasted_lens.insert(l0_broadcasted_lens.end(), l0_it, s0_lens.end());
-                l1_broadcasted_lens = output_lens;
-                l1_broadcasted_lens.insert(l1_broadcasted_lens.end(), l1_it, s1_lens.end());
-                if(s0_lens != l0_broadcasted_lens)
-                {
-                    ba0 = info.add_instruction(
-                        make_op("multibroadcast", {{"out_lens", l0_broadcasted_lens}}), a0);
-                }
-                if(s1_lens != l1_broadcasted_lens)
-                {
-                    ba1 = info.add_instruction(
-                        make_op("multibroadcast", {{"out_lens", l1_broadcasted_lens}}), a1);
-                }
-            }
+                set_bias_arg(info, args, 2, a0, ba0);
+                set_bias_arg(info, args, 3, a1, ba1);
```

Comment:
`ba0` and `ba1` are used for broadcasted `a0` and `a1`. 
Do not mix them with bias_arg. 

You can make `set_bias_arg` to return `instruction_ref`

Reply:
Will do